proxy/proxy.go

```diff
@@ -1443,6 +1444,9 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		r.URL.Path = rfc.PatchPath(r.URL.Path, r.URL.RawPath)
 	}
 
+	// Change /foo/../bar to /bar for matching and passing upstream
+	r.URL.Path = httppath.Clean(r.URL.Path)
```

Comment:
Isn't this a breaking change? Should it be feature option maybe?

Reply:
maybe, although there is a problem for my team accessing these kinda of feature flags. Also, one could also argue this is a bug fix and therefore everyone should get it.

Reply:
the alternative would be to use an explicit cleanPath() filter, but I am not aware of a scenario that would get broken this way.

Reply:
An explicit `cleanPath()` filter is probably the way to go. I don't expect a proxy to mutate my request unless explicitly expressed with a filter on the route.